### PR TITLE
ubuntu 20 ansible workaround

### PR DESCRIPTION
**Pull request by jvonau**
### Fixes Bug
#2270 
### Description of changes proposed in this pull request.
untested more notes than working code

### CODE REVIEW

**Comment by jvonau** [comment]
Now tested on raspbian OS works as expected installing the key and ppa. 2.9.4 should be installed on Ubuntu-20.04 without installing the repo key and ppa for ansible.

File: scripts/ansible [hunk 1]
```diff
@@ -64,16 +64,17 @@ elif [ -f /etc/centos-release ]; then
...
+        apt -y install dirmngr
+        echo "deb http://ppa.launchpad.net/ansible/ansible/ubuntu bionic main" \
          > /etc/apt/sources.list.d/iiab-ansible.list
 
-    echo -e '\nIF YOU FACE ERROR "signatures couldn'"'"'t be verified because the public key is not available" THEN REPEATEDLY RE-RUN "sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 93C4A3FD7BB9C367"\n'
-    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 93C4A3FD7BB9C367
-
+        echo -e '\nIF YOU FACE ERROR "signatures couldn'"'"'t be verified because the public key is not available" THEN REPEATEDLY RE-RUN "sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 93C4A3FD7BB9C367"\n'
+        apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 93C4A3FD7BB9C367
+    fi
```
Comment (holta) [inline scripts/ansible:81]: Can ` apt install ansible` be considered on Ubuntu 20.04, perhaps with an else clause here?

File: scripts/ansible [hunk 2]
```diff
@@ -91,12 +92,11 @@ elif [ -f /etc/debian_version ]; then    # Includes Debian, Ubuntu & Raspbian
     #wget http://download.iiab.io/packages/ansible_2.4.2.0-1ppa~xenial_all.deb
     #apt -y --allow-downgrades install ./ansible_2.4.2.0-1ppa~xenial_all.deb
 
-    echo -e 'PPA source "deb http://ppa.launchpad.net/ansible/ansible/ubuntu bionic main"'
-    echo -e "successfully saved to /etc/apt/sources.list.d/iiab-ansible.list\n"
-
-    echo -e "IF *OTHER* ANSIBLE SOURCES APPEAR BELOW, PLEASE MANUALLY REMOVE THEM TO"
-    echo -e 'ENSURE ANSIBLE UPDATES CLEANLY: (then re-run this script to be sure!)\n'
-    grep '^deb .*ansible' /etc/apt/sources.list /etc/apt/sources.list.d/*.list | grep -v '^/etc/apt/sources.list.d/iiab-ansible.list:' || true    # Override bash -e (instead of aborting at 1st error)
+#    echo -e 'PPA source "deb http://ppa.launchpad.net/ansible/ansible/ubuntu bionic main"'
```
Comment (holta) [inline scripts/ansible]: Lines 95-99 help people who've ended up putting too many different Ansible sources into /etc/apt/sources.list.d/*.list — giving them important guidance on how to get back on track.

So can we move these 5 lines, so they live just prior to the `fi` on Line 77 of this PR?

File: scripts/ansible (hunk 1 above)
Comment (jvonau) [reply scripts/ansible:81]: Why? The PPA and required key are not being used to install, ansible is being provide by Ubuntu, just setting up the PPA and key are skipped when 'focal' is matched. What do you propose?

File: scripts/ansible (hunk 2 above)
Comment (jvonau) [reply scripts/ansible]: ok prior to 77 works, was looking to limit misleading information.

File: scripts/ansible (hunk 1 above)
Comment (holta) [reply scripts/ansible:81]: I was assuming @georgejhunt manually ran something like `apt install ansible` on Ubuntu 20.04 pre-releases, and that scripts/ansible here could emulate his approach?

(But perhaps I have the details wrong...and there's a better way?)

Comment (jvonau) [reply scripts/ansible:81]: Think that might be the case and perhaps left out the ansible dependencies, only to install them later in roles/...

**Comment by holta** [comment]
@jvonau smoke-tested this on RPi 4 with both OS's: Ubuntu 20.04 daily pre-release & Raspbian.

Thanks also to @georgejhunt who informed this PR as it evolved.

**Pull request merged**